### PR TITLE
Drop job module needles table

### PR DESCRIPTION
**Pull request by coolo**
See https://progress.opensuse.org/issues/34531

### CODE REVIEW

File: lib/OpenQA/Schema/Result/JobModules.pm [hunk 1]
```diff
@@ -83,14 +83,27 @@ __PACKAGE__->belongs_to(
...
 
+# override to straighten out needle references
+sub delete {
+    my ($self) = @_;
+
+    my $schema = $self->result_source->schema;
+
+    my $needles = $schema->resultset('Needles');
+
+    $needles->search({last_seen_module_id    => $self->id})->update({last_seen_module_id    => undef});
```
Comment (Martchus) [inline lib/OpenQA/Schema/Result/JobModules.pm]: So you need this despite the `on_delete => 'SET NULL'` above?  
(same for the next line)

**Comment by Martchus** [comment]
Looks good beside the one question. I tested it locally and the migration seems to work.

File: lib/OpenQA/Schema/Result/JobModules.pm (hunk 1 above)
Comment (coolo) [reply lib/OpenQA/Schema/Result/JobModules.pm]: Yes. I think the on_delete is toothless

Comment (coolo) [reply lib/OpenQA/Schema/Result/JobModules.pm]: I tried without the function now - possibly my schema wasn't properly setup when I tried

**Comment by coolo** [comment]
question clarified - target coverage rather mystic...

**Pull request merged**